core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -120,6 +120,8 @@ private TableProperties() {
   public static final String METRICS_MODE_COLUMN_CONF_PREFIX = "write.metadata.metrics.column.";
   public static final String DEFAULT_WRITE_METRICS_MODE = "write.metadata.metrics.default";
   public static final String DEFAULT_WRITE_METRICS_MODE_DEFAULT = "truncate(16)";
+  public static final String SORTED_COL_DEFAULT_METRICS_MODE = "write.metadata.sorted.metrics.default";
```

Comment:
Actually, I think we should simplify it a bit. The use case I was talking about is when the user configures the default sort mode as `none` or `counts` but creates a table with a sort order. In that case, we should promote the metrics for sort columns to be at least `truncate(16)` unless the user sets a mode for sort columns explicitly. It is probably too dangerous to promote to `full` as the values may be too long. I guess `truncate(16)` is a reasonable default for sort columns and it will apply only to sort and binary columns. Longs/integers are not affected.

Internally, we may want to change the default value to `counts` instead of `truncate(16)` for tables with many columns as we have a lot of tables with 100+ columns and we don't want to a ton of unnecessary metadata. But I am not sure that the community wants to do the same.

Reply:
Thoughts, @szehon-ho @RussellSpitzer @rdblue?

Reply:
I'd also say the promotion should be implicit and we don't need extra table properties.

Reply:
I think i'm also in favor of just making the change without a flag to configure. Just to make configuration have one parameter less, do we really have a lot of use cases where truncate(16) isn't the right thing to do? And if it isn't we can just let users manually reconfigure those columns individually if they need.

Reply:
@szehon-ho, could you update the PR to match the behavior in [this](https://github.com/apache/iceberg/pull/2240#discussion_r577899216) comment, please?

Reply:
Yes I'll take a look, thanks for the feedback guys

Reply:
Recap if I understand correctly: if default < truncate(16), then sort columns => truncate(16).  And no table property (user overrides the sort col metric property itself)

Reply:
That’s just my suggestion, I can always be convinced otherwise too 😊

Reply:
No you guys have a point, full is unnecessary if someone sorts by a huge text for example, and there's already a good way to override the config.  Updated the PR and changed the title to reflect.

Reply:
I agree with this behavior and I also would not add a new table property.